### PR TITLE
dialect/sql/sqlgraph: support sql.Scanner types when scanning IDs

### PR DESCRIPTION
**Pull request by a8m**
Fixed https://github.com/ent/ent/issues/1985

### CODE REVIEW

File: dialect/sql/sqlgraph/graph.go [hunk 1]
```diff
@@ -1301,6 +1301,11 @@ func (c *creator) insertLastID(ctx context.Context, insert *sql.InsertBuilder) e
 		}
 		defer rows.Close()
 		if !c.ID.Type.Numeric() {
+			// If the ID implements the sql.Scanner
```
Comment (masseelch) [inline dialect/sql/sqlgraph/graph.go:1304]: Just out of curiosity: What about numeric types that have a custom `sql.Sanner` implementation?

Comment (a8m) [reply dialect/sql/sqlgraph/graph.go:1304]: Good point. The `c.ID.Value.(field.ValueScanner)` check needs to move one level up. 

Thanks 🙏

File: entc/gen/template/dialect/sql/create.tmpl [hunk 2]
```diff
@@ -21,7 +21,12 @@ func ({{ $receiver }} *{{ $builder }}) sqlSave(ctx context.Context) (*{{ $.Name
 	}
 	{{- if and $.ID.UserDefined (not $.ID.Type.Numeric) }}
```
Comment (masseelch) [inline entc/gen/template/dialect/sql/create.tmpl:22]: Same on this line, isn't it?

**Comment by a8m** [comment]
Thanks @masseelch for the review ❤️

**Pull request merged**